Namespace(batch_size=2000, cpuonly=False, dataset='imagenet', downloadall=False, end=10000, freeze=True, freeze_idx=21, manual_split=False, model='resnet50', num_epochs=1, sequential=False, split_idx=100, start=0, testonly=False, use_intermediate=False)
==> Preparing data..
==> Building model..
Using split index: 100
Freezing the lower layers of the model (resnet50) till index 21
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Read 254.37291526794434 MBs for this batch
Streaming imagenet data took 14.510488271713257 seconds

Epoch: 0
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Memory occpied: (1222.0, 3.0)
Time of next(dataloader) is: 7.653839349746704
Memory occpied: (1222.0, 3.0)
Time for copying to cuda: 0.3241090774536133
Memory occpied: (2372.0, 266.0)
Memory occpied: (2372.0, 586.0)
Memory occpied: (2372.0, 880.0)
Memory occpied: (12516.0, 11904.0)
Memory occpied: (12490.0, 11904.0)
Memory occpied: (2390.0, 11904.0)
Read 246.17637538909912 MBs for this batch
Streaming imagenet data took 17.351431608200073 seconds
Memory occpied: (2390.0, 14620.0)
Memory occpied: (2390.0, 12286.0)
Memory occpied: (2390.0, 8844.0)
Memory occpied: (2390.0, 8078.0)
Memory occpied: (2390.0, 9972.0)
Memory occpied: (2390.0, 11440.0)
Exception: Caught RuntimeError in replica 0 on device 0.
Original Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/parallel/parallel_apply.py", line 61, in _worker
    output = module(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torchvision/models/resnet.py", line 249, in forward
    return self._forward_impl(x)
  File "/usr/local/lib/python3.8/dist-packages/torchvision/models/resnet.py", line 237, in _forward_impl
    x = self.layer1(x)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/container.py", line 141, in forward
    input = module(input)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torchvision/models/resnet.py", line 136, in forward
    identity = self.downsample(x)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/container.py", line 141, in forward
    input = module(input)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/batchnorm.py", line 168, in forward
    return F.batch_norm(
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/functional.py", line 2282, in batch_norm
    return torch.batch_norm(
RuntimeError: CUDA out of memory. Tried to allocate 2.99 GiB (GPU 0; 14.76 GiB total capacity; 7.95 GiB already allocated; 1.89 GiB free; 11.77 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

The whole process took 45.34745693206787 seconds
